librz/bin/bfile.c

```diff
@@ -110,257 +107,118 @@ static void print_string(RzBinFile *bf, RzBinString *string, int raw, PJ *pj) {
 	}
 }
 
+static inline void detected_string_to_bin_string(RzBinString *dst, const RzDetectedString *src) {
+	int type = -1;
+	switch (src->type) {
+	case RZ_STRING_ENC_LATIN1:
+		type = RZ_STRING_TYPE_ASCII;
+		break;
+	case RZ_STRING_ENC_UTF8:
+		type = RZ_STRING_TYPE_UTF8;
+		break;
+	case RZ_STRING_ENC_UTF16LE:
+		type = RZ_STRING_TYPE_WIDE;
+		break;
+	case RZ_STRING_ENC_UTF32LE:
+		type = RZ_STRING_TYPE_WIDE32;
+		break;
+	case RZ_STRING_ENC_UTF16BE:
+		type = RZ_STRING_TYPE_WIDE_BE;
+		break;
+	case RZ_STRING_ENC_UTF32BE:
+		type = RZ_STRING_TYPE_WIDE32_BE;
+		break;
+	case RZ_STRING_ENC_GUESS:
+		type = RZ_STRING_TYPE_DETECT;
+		break;
+	default:
+		break;
+	}
+
+	dst->string = rz_str_ndup(src->string, src->size);
```

Comment:
I think inside this function we should free the `src` string and re-use the src->string from there instead of duplicating it. There are already a lot of memory allocations in Rizin and many of them can probably be avoided.

Reply:
Modified